spec/api/goals_api_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+describe '/api/goals' do
+
+  context 'GET index' do
+
+    context 'with correct credentials' do
+      it 'returns a list of current_user goals as JSON' do
+        user = create(:user)
+        # goal_1 = create(:goal, user: user)
+        # goal_2 = create(:goal, user: user)
+        # goal_3 = create(:goal, user: user)
+        goals = create_list(:goal, 3, user: user)
+        headers = json_request_headers
+        headers['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials(user.email,user.password)
```

Comment:
@jeffminnear `headers['HTTP_AUTHORIZATION']` can be safely replaced with `headers['Authorization']` to make the spec more closely imitate what a real client like curl does for HTTP Basic Authentication with username and password. 

```bash
# Sends a request to the given URL that includes a request header named
# "Authorization"
curl -u a@b.c:helloworld http://localhost:3000/api/users/
```
Sample Authorization header can be seen when using the `-v` (verbose) option with curl, e.g.:
```bash
curl -v -u a@b.c:helloworld http://localhost:3000/api/users/
```
outputs a lot of data, including the request headers:

```
...
* Server auth using Basic with user 'a@b.c'
> GET /api/users/ HTTP/1.1
> Authorization: Basic YUBiLmM6aGVsbG93b3JsZA==
> User-Agent: curl/7.30.0
> Host: localhost:3000
> Accept: */*
> 
...
```